File: LeetCode/0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int l = 0;

        for (int i = 0; i < t.length()&& l < s.length(); i++){
            if(s.charAt(l) == t.charAt(i)){
                l++;
            }
        }
        return l == s.length();
    }
}

File: LeetCode/0392-is-subsequence/0392-is-subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        var l = 0
        var i = 0
        while (i < t.length && l < s.length) {
            if (s.get(l) == t.get(i)) l++
            i++
        }
        return l == s.length
    }
}